lib/rubygems/basic_specification.rb

```diff
@@ -175,11 +175,13 @@ def to_fullpath(path)
         begin
           fullpath = nil
           suffixes = Gem.suffixes
-          suffixes.find do |suf|
+          if suffixes.find do |suf|
             full_require_paths.find do |dir|
               File.file?(fullpath = "#{dir}/#{path}#{suf}")
             end
-          end ? fullpath : nil
+          end
+            fullpath
+          end
```

Comment:
This code is a bit crazy, no?

I think if we extract the logic to a separate method we can make it simpler, like

```ruby
def find_full_path(path)
  Gem.suffixes.each do |suf|
    full_require_paths.each do |dir|
      fullpath = "#{dir}/#{path}#{suf}"
      return fullpath if File.file?(fullpath)
    end
  end

  nil
end
```

Reply:
Maked sense. I rewrite it to this:

```ruby
  def to_fullpath(path)
    if activated?
      @paths_map ||= {}
      Gem.suffixes.each do |suf|
        full_require_paths.each do |dir|
          fullpath = "#{dir}/#{path}#{suf}"
          next unless File.file?(fullpath)
          @paths_map[path] ||= fullpath
        end
      end
      @paths_map[path]
    end
  end
```

Reply:
Oh, that looks great! 👍